presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/MultinodeTlsKerberos.java

```diff
@@ -0,0 +1,101 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests.product.launcher.env.environment;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.tests.product.launcher.PathResolver;
+import io.prestosql.tests.product.launcher.docker.DockerFiles;
+import io.prestosql.tests.product.launcher.env.DockerContainer;
+import io.prestosql.tests.product.launcher.env.Environment;
+import io.prestosql.tests.product.launcher.env.EnvironmentOptions;
+import io.prestosql.tests.product.launcher.env.common.AbstractEnvironmentProvider;
+import io.prestosql.tests.product.launcher.env.common.Hadoop;
+import io.prestosql.tests.product.launcher.env.common.Kerberos;
+import io.prestosql.tests.product.launcher.env.common.Standard;
+import org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import javax.inject.Inject;
+
+import java.io.File;
+import java.time.Duration;
+import java.util.Objects;
+
+import static com.google.common.base.Verify.verify;
+import static io.prestosql.tests.product.launcher.env.common.Hadoop.CONTAINER_PRESTO_HIVE_PROPERTIES;
+import static io.prestosql.tests.product.launcher.env.common.Standard.CONTAINER_PRESTO_CONFIG_PROPERTIES;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.containers.BindMode.READ_ONLY;
+
+public class MultinodeTlsKerberos
+        extends AbstractEnvironmentProvider
+{
+    private final PathResolver pathResolver;
+    private final DockerFiles dockerFiles;
+
+    private final String prestoDockerImageName;
+    private final File serverPackage;
+
+    @Inject
+    public MultinodeTlsKerberos(
+            PathResolver pathResolver,
+            DockerFiles dockerFiles,
+            Standard standard,
+            Hadoop hadoop,
+            Kerberos kerberos,
+            EnvironmentOptions environmentOptions)
+    {
+        super(ImmutableList.of(standard, hadoop, kerberos));
+        this.pathResolver = requireNonNull(pathResolver, "pathResolver is null");
+        this.dockerFiles = requireNonNull(dockerFiles, "dockerFiles is null");
+        String hadoopBaseImage = requireNonNull(environmentOptions.hadoopBaseImage, "environmentOptions.hadoopBaseImage is null");
+        String imagesVersion = requireNonNull(environmentOptions.imagesVersion, "environmentOptions.imagesVersion is null");
+        prestoDockerImageName = hadoopBaseImage + "-kerberized:" + imagesVersion;
+        serverPackage = requireNonNull(environmentOptions.serverPackage, "environmentOptions.serverPackage is null");
+    }
+
+    @Override
+    @SuppressWarnings("resource")
+    protected void extendEnvironment(Environment.Builder builder)
+    {
+        super.extendEnvironment(builder);
```

Comment:
Not specific to this PR, but this pattern is strange. Why do we call an empty super method in the abstract parent class (which is designed for subclasses to implement)?

Reply:
right. Let's fix them everywhere consistently

Reply:
Done